qutip/lattice.py

```diff
@@ -0,0 +1,814 @@
+    val_t : list of str
+        The second list of str's specifying the sites/degrees of freedom in the
+        unitcell
+
+    limit:  list
+        Informs the user of exceeding the capacity of the cell_structures
+        function.
+
+    Returns
+    -------
+    cell_H : numpy ndarray
+        tensor structure of the cell Hamiltonian elements
+    inter_cell_T : numpy ndarray
+        tensor structure of the inter cell Hamiltonian elements
+    """
+    SN = len(val_s)
+    TN = len(val_t)
+    LM = len(limit)
+
+    if (SN * TN == 0 and SN != 0):
+        scell_H = [[[] for i in range(SN)] for j in range(SN)]
+        sinter_cell_T = [[[] for i in range(SN)] for j in range(SN)]
+
+        for ir in range(SN):
+            for ic in range(SN):
+                sst = val_s[ir]+" H "+val_s[ic]
+                scell_H[ir][ic] = "<" + sst + ">"
+                sinter_cell_T[ir][ic] = "<cell(i):" + sst + ":cell(i+1) >"
+
+    if (SN * TN * LM == 0 and SN * TN != 0):
+        scell_H = [[[] for i in range(SN * TN)] for j in range(SN * TN)]
+        sinter_cell_T = [[[] for i in range(SN * TN)] for j in range(SN * TN)]
+
+        for ir in range(SN):
+            for jr in range(TN):
+                for ic in range(SN):
+                    for jc in range(TN):
+
+                        sst = []
+                        sst.append(val_s[ir])
+                        sst.append(val_t[jr])
+                        sst.append(" H ")
+                        sst.append(val_s[ic])
+                        sst.append(val_t[jc])
+                        sst = ''.join(sst)
+
+                        scell_H[ir * TN+jr][ic * TN+jc] = "<" + sst + ">"
+                        llt = []
+                        llt.append("<cell(i):")
+                        llt.append(sst)
+                        llt.append(":cell(i+1) >")
+                        sinter_cell_T[ir * TN+jr][ic * TN+jc] = llt
+
+    if (SN*TN*LM != 0):
+        print("The cell_structures() can not handle more than 2 lists!")
+        scell_H = []
+        sinter_cell_T = []
+
+    cell_H = np.zeros(np.shape(scell_H), dtype=complex)
+    inter_cell_T = np.zeros(np.shape(sinter_cell_T), dtype=complex)
+    return (scell_H, sinter_cell_T, cell_H, inter_cell_T)
+
+
+class Lattice1d():
+    """A class for representing a 1d crystal.
+
+    The Lattice1d class can be defined with any specific unit cells and a
+    specified number of unit cells in the crystal. It can return dispersion
+    relationship, position operators, Hamiltonian in the positio represention
+    etc.
+
+    Parameters
+    ----------
+    num_cell : int
+        The number of cells in the crystal.
+    boundary : str
+        Specification of the type of boundary the crystal is defined with.
+    cell_num_site : int
+        The number of sites in the unit cell.
+    cell_site_dof : list of int/ int
+        The tensor structure  of the degrees of freedom at each site of a unit
+        cell.
+    cell_Hamiltonian :  Qobj
```

Comment:
Change Qobj to :class:`qutip.Qobj` so that Sphinx can add a link in the documentation